src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -250,29 +251,28 @@ public void run() {
     driver.run();
   }
 
-  private boolean launchNode(SchedulerDriver driver, Offer offer,
-        String nodeName, List<String> taskTypes, String executorName) {
+  private void launchNode(SchedulerDriver driver, Offer offer,
+        String nodeName, Map<String,String> taskInfo, String executorName) {
     //nodeName is the type of executor to launch
     //executorName is to distinguish different types of nodes
-    //taskType is the type of task in mesos to launch on the node
-    //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
+    //taskInfo is taskType mapped to taskName
     log.info(String.format("Launching node of type %s with tasks %s", nodeName,
-        taskTypes.toString()));
+        taskInfo));
     String taskIdName = String.format("%s.%s.%d", nodeName, executorName,
         System.currentTimeMillis());
     List<Resource> resources = getExecutorResources();
     ExecutorInfo executorInfo = createExecutor(taskIdName, nodeName, executorName, resources);
     List<TaskInfo> tasks = new ArrayList<>();
-    for (String taskType : taskTypes) {
+    for (Map.Entry<String,String> taskItem : taskInfo.entrySet()) {
+      String taskType = taskItem.getKey();
```

Comment:
Please create
EntrySet taskInfos = taskInfo.entrySet();
for (Map.Entry<String,String> taskItem : taskInfos)

Sorry that you disagree, but as the project owner I have the final say on style here and we need to be consistent with Java style patterns. As mentioned before,one big reason we avoid method chaining is for clarity and for better error handling. If an error is thrown on a specific line, we will know the source.